lib/KTable/KTable.vue

```diff
@@ -0,0 +1,234 @@
+          <KTableGridItem
+            v-for="(cell, cellIndex) in row"
+            :key="cellIndex"
+            :content="cell"
+            :align="headers[cellIndex].dataType === 'numeric' ? 'right' : 'left'"
+            :rowIndex="rowIndex"
+            :cellIndex="cellIndex"
+            @keydown="handleKeydown($event, rowIndex, cellIndex)"
+          />
+        </tr>
+      </tbody>
+    </table>
+  </div>
+
+</template>
+
+
+<script>
+
+  import KTableGridItem from './KTableGridItem.vue';
+
+  export default {
+    name: 'KTable',
+    components: {
+      KTableGridItem,
+    },
+    props: {
+      headers: {
+        type: Array,
+        required: true,
+      },
+      rows: {
+        type: Array,
+        required: true,
+      },
+      caption: {
+        type: String,
+        required: false,
+        default: null,
+      },
+    },
+    data() {
+      return {
+        sortKey: null,
+        sortOrder: 'asc',
+      };
+    },
+    computed: {
+      sortedRows() {
+        if (this.sortKey === null) return this.rows;
+
+        const sortedRows = [...this.rows];
+        const sortHeader = this.headers[this.sortKey];
+
+        if (sortHeader.dataType === 'numeric') {
+          sortedRows.sort((a, b) => a[this.sortKey] - b[this.sortKey]);
+        } else if (sortHeader.dataType === 'string' || sortHeader.dataType === 'date') {
+          sortedRows.sort((a, b) => a[this.sortKey].localeCompare(b[this.sortKey]));
+        }
+
+        if (this.sortOrder === 'desc') {
+          sortedRows.reverse();
+        }
+
+        return sortedRows;
+      },
+    },
+    methods: {
+      handleSort(index) {
+        const clickedElement = event.target.tagName.toLowerCase();
+        if (clickedElement !== 'th' || this.headers[index].dataType === 'others') return;
+
+        if (this.sortKey === index) {
+          this.sortOrder = this.sortOrder === 'asc' ? 'desc' : 'asc';
+        } else {
+          this.sortKey = index;
+          this.sortOrder = 'asc';
+        }
+      },
+      handleKeydown(event, rowIndex, cellIndex) {
+        const key = event.key;
+        const totalRows = this.rows.length;
+        const totalCols = this.headers.length;
+
+        let nextRowIndex = rowIndex;
+        let nextCellIndex = cellIndex;
+
+        switch (key) {
+          case 'ArrowUp':
+            if (rowIndex === 0) {
+              this.focusHeader(cellIndex);
+              return;
+            } else if (rowIndex > 0) {
+              nextRowIndex = rowIndex - 1;
+            } else {
+              return;
+            }
+            break;
+          case 'ArrowDown':
+            if (rowIndex === -1) {
+              nextRowIndex = 0;
+            } else {
+              nextRowIndex = (rowIndex + 1) % totalRows;
+            }
+            break;
+          case 'ArrowLeft':
+            if (rowIndex === -1) {
+              nextCellIndex = cellIndex > 0 ? cellIndex - 1 : totalCols - 1;
```

Comment:
It seems to me that `cellIndex` is holding the column index for the cell, is that right? If so, I'd recommend renaming all related variables to `columnIndex`, `nextColumnIndex`, etc. or something similar to help a bit with clarity.

Reply:
Okayy!